osu.Game/Online/Chat/MessageNotifier.cs

```diff
@@ -0,0 +1,194 @@
+
+        private bool checkForPMs(Channel channel, Message message)
+        {
+            if (!notifyOnPM.Value || channel.Type != ChannelType.PM)
+                return false;
+
+            var notification = new PrivateMessageNotification(message.Sender.Username, channel);
+
+            notificationOverlay?.Post(notification);
+
+            return true;
+        }
+
+        private void checkForMentions(Channel channel, Message message, string username)
+        {
+            if (!notifyOnMention.Value || !isMentioning(message.Content, username))
+                return;
+
+            var notification = new MentionNotification(message.Sender.Username, channel);
+            notificationOverlay?.Post(notification);
+        }
+
+        /// <summary>
+        /// Checks if <paramref name="message"/> contains <paramref name="username"/>, if not, retries making spaces into underscores.
+        /// </summary>
+        /// <returns>If the <paramref name="message"/> mentions the <paramref name="username"/></returns>
+        private static bool isMentioning(string message, string username) => message.IndexOf(username, StringComparison.OrdinalIgnoreCase) != -1 || message.IndexOf(username.Replace(' ', '_'), StringComparison.OrdinalIgnoreCase) != -1;
+
+        public class PrivateMessageNotification : SimpleNotification
+        {
+            public PrivateMessageNotification(string username, Channel channel)
+            {
+                Icon = FontAwesome.Solid.Envelope;
+                Text = $"You received a private message from '{username}'. Click to read it!";
+                this.channel = channel;
+            }
+
+            private readonly Channel channel;
+
+            public override bool IsImportant => false;
+
+            [BackgroundDependencyLoader]
+            private void load(OsuColour colours, ChatOverlay chatOverlay, NotificationOverlay notificationOverlay, ChannelManager channelManager)
+            {
+                IconBackgound.Colour = colours.PurpleDark;
+
+                Activated = delegate
+                {
+                    notificationOverlay.Hide();
+                    chatOverlay.Show();
+                    channelManager.CurrentChannel.Value = channel;
+
+                    return true;
+                };
+            }
+        }
+
+        public class MentionNotification : SimpleNotification
+        {
+            public MentionNotification(string username, Channel channel)
+            {
+                Icon = FontAwesome.Solid.At;
+                Text = $"Your name was mentioned in chat by '{username}'. Click to find out why!";
+                this.channel = channel;
+            }
+
+            private readonly Channel channel;
+
+            public override bool IsImportant => false;
+
+            [BackgroundDependencyLoader]
+            private void load(OsuColour colours, ChatOverlay chatOverlay, NotificationOverlay notificationOverlay, ChannelManager channelManager)
```

Comment:
Probably not a big deal, but code duplication can be reduced by adding a common base type here (like `OpenChannelNotification`).